Mathlib/Topology/Algebra/ContinuousMonoidHom.lean

```diff
@@ -413,3 +415,280 @@ theorem locallyCompactSpace_of_hasBasis (V : ℕ → Set Y)
+  e.toEquiv.apply_symm_apply y
+
+/-- `e.symm` is a left inverse of `e`, written as `e.symm (e y) = y`. -/
+@[to_additive (attr := simp) "`e.symm` is a left inverse of `e`, written as `e.symm (e y) = y`."]
+theorem symm_apply_apply (e : M ≃ₜ* N) (x : M) : e.symm (e x) = x :=
+  e.toEquiv.symm_apply_apply x
+
+@[to_additive (attr := simp)]
+theorem symm_comp_self (e : M ≃ₜ* N) : e.symm ∘ e = id :=
+  funext e.symm_apply_apply
+
+@[to_additive (attr := simp)]
+theorem self_comp_symm (e : M ≃ₜ* N) : e ∘ e.symm = id :=
+  funext e.apply_symm_apply
+
+@[to_additive]
+theorem apply_eq_iff_symm_apply (e : M ≃ₜ* N) {x : M} {y : N} : e x = y ↔ x = e.symm y :=
+  e.toEquiv.apply_eq_iff_eq_symm_apply
+
+@[to_additive]
+theorem symm_apply_eq (e : M ≃ₜ* N) {x y} : e.symm x = y ↔ x = e y :=
+  e.toEquiv.symm_apply_eq
+
+@[to_additive]
+theorem eq_symm_apply (e : M ≃ₜ* N) {x y} : y = e.symm x ↔ e y = x :=
+  e.toEquiv.eq_symm_apply
+
+@[to_additive]
+theorem eq_comp_symm {α : Type*} (e : M ≃ₜ* N) (f : N → α) (g : M → α) :
+    f = g ∘ e.symm ↔ f ∘ e = g :=
+  e.toEquiv.eq_comp_symm f g
+
+@[to_additive]
+theorem comp_symm_eq {α : Type*} (e : M ≃ₜ* N) (f : N → α) (g : M → α) :
+    g ∘ e.symm = f ↔ g = f ∘ e :=
+  e.toEquiv.comp_symm_eq f g
+
+@[to_additive]
+theorem eq_symm_comp {α : Type*} (e : M ≃ₜ* N) (f : α → M) (g : α → N) :
+    f = e.symm ∘ g ↔ e ∘ f = g :=
+  e.toEquiv.eq_symm_comp f g
+
+@[to_additive]
+theorem symm_comp_eq {α : Type*} (e : M ≃ₜ* N) (f : α → M) (g : α → N) :
+    e.symm ∘ g = f ↔ g = e ∘ f :=
+  e.toEquiv.symm_comp_eq f g
+
+end symm
+
+section trans
+
+variable {L : Type*} [Mul L] [TopologicalSpace L]
+
+/-- The composition of two ContinuousMulEquiv. -/
+@[to_additive "The composition of two ContinuousAddEquiv."]
+def trans (cme1 : M ≃ₜ* N) (cme2 : N ≃ₜ* L) : M ≃ₜ* L :=
+  { cme1.toMulEquiv.trans cme2.toMulEquiv with
+  continuous_toFun := by convert Continuous.comp cme2.continuous_toFun cme1.continuous_toFun
+  continuous_invFun := by convert Continuous.comp cme1.continuous_invFun cme2.continuous_invFun }
+
+@[to_additive (attr := simp)]
+theorem coe_trans (e₁ : M ≃ₜ* N) (e₂ : N ≃ₜ* L) : ↑(e₁.trans e₂) = e₂ ∘ e₁ := rfl
+
+@[to_additive (attr := simp)]
+theorem trans_apply (e₁ : M ≃ₜ* N) (e₂ : N ≃ₜ* L) (m : M) : e₁.trans e₂ m = e₂ (e₁ m) := rfl
+
+@[to_additive (attr := simp)]
+theorem symm_trans_apply (e₁ : M ≃ₜ* N) (e₂ : N ≃ₜ* L) (l : L) :
+    (e₁.trans e₂).symm l = e₁.symm (e₂.symm l) := rfl
+
+@[to_additive (attr := simp)]
+theorem symm_trans_self (e : M ≃ₜ* N) : e.symm.trans e = refl N :=
+  DFunLike.ext _ _ e.apply_symm_apply
+
+@[to_additive (attr := simp)]
+theorem self_trans_symm (e : M ≃ₜ* N) : e.trans e.symm = refl M :=
+  DFunLike.ext _ _ e.symm_apply_apply
+
+end trans
+
+section unique
+
+/-- The `MulEquiv` between two monoids with a unique element. -/
+@[to_additive "The `AddEquiv` between two `AddMonoid`s with a unique element."]
+def continuousMulEquivOfUnique {M N} [Unique M] [Unique N] [Mul M] [Mul N]
```

Comment:
I wonder if we should rename all `MyType.myTypeOfUnique` to `MyType.ofUnique`, but this is out of scope for this PR.

Reply:
I've opened #20243

Reply:
#20243 is now merged. Can you update this PR please, @Thmoas-Guan ?

Reply:
Updated, thanks!